cats/shared/src/main/scala/org/atnos/eff/addon/cats/effect/IOEffect.scala

```diff
@@ -72,6 +74,58 @@ trait IOInterpretation extends IOTypes {
           io.attempt
       })
   }
+
+  /** memoize the io result using a cache */
+  def memoize[A](key: AnyRef, cache: Cache, io: IO[A]): IO[A] =
+    cache.get[A](key).fold(io.map { r => cache.put(key, r); r })(IO.pure)
+
+  /**
+    * Memoize io effects using a cache
+    *
+    * if this method is called with the same key the previous value will be returned
+    */
+  def ioMemo[R, A](key: AnyRef, cache: Cache, e: Eff[R, A])(implicit task: IO /= R): Eff[R, A] =
+    ioAttempt(Eff.memoizeEffect(e, cache, key)).flatMap {
+      case Left(t) => Eff.send(taskSequenceCached.reset(cache, key)) >> IOEffect.ioRaiseError(t)
+      case Right(a) => Eff.pure(a)
+    }
+
+  /**
+    * Memoize task values using a memoization effect
+    *
+    * if this method is called with the same key the previous value will be returned
+    */
+  def ioMemoized[R, A](key: AnyRef, e: Eff[R, A])(implicit task: IO /= R, m: Memoized |= R): Eff[R, A] =
+    MemoEffect.getCache[R].flatMap(cache => ioMemo(key, cache, e))
+
+  def runIoMemo[R, U, A](cache: Cache)(effect: Eff[R, A])(implicit m: Member.Aux[Memoized, R, U], task: IO |= U): Eff[U, A] = {
+    interpret.translate(effect)(new Translate[Memoized, U] {
+      def apply[X](mx: Memoized[X]): Eff[U, X] =
+        mx match {
+          case Store(key, value) => IOEffect.ioDelay(cache.memo(key, value()))
+          case GetCache() => IOEffect.ioDelay(cache)
+        }
+    })
+  }
+
+  implicit val taskSequenceCached: SequenceCached[IO] = new SequenceCached[IO] {
```

Comment:
This should be called `ioSequenceCached` isn't it?

Reply:
yep :)